diff -uNr wvdial-1.41/CHANGES wvdial-libs/CHANGES
--- wvdial-1.41/CHANGES	Thu Jan  6 21:57:05 2000
+++ wvdial-libs/CHANGES	Fri Jan 28 15:15:50 2000
@@ -2,7 +2,7 @@
 Changes to WvDial since the initial release:
 
 
-1.41 (1999 01 06)
+1.41 (2000 01 06)
 ----
  - Some internal changes to wvstreams for compatibility with non-gcc
 	compilers.  Nothing much that will interest most people, but anyone
diff -uNr wvdial-1.41/configfile/wvconf.h wvdial-libs/configfile/wvconf.h
--- wvdial-1.41/configfile/wvconf.h	Sun Nov 28 14:39:33 1999
+++ wvdial-libs/configfile/wvconf.h	Fri Jan 28 15:15:50 2000
@@ -85,7 +85,7 @@
 class WvConf : public WvConfigSectionList
 {
 public:
-    WvConf(const WvString &_filename, int _create_mode = 0666);
+    WvConf(const WvString &_filename, int _create_mode = 0600);
     ~WvConf();
     
     bool isok() const
diff -uNr wvdial-1.41/rules.mk wvdial-libs/rules.mk
--- wvdial-1.41/rules.mk	Wed Oct 20 18:37:00 1999
+++ wvdial-libs/rules.mk	Fri Jan 28 15:15:50 2000
@@ -71,8 +71,8 @@
 #
 # Typical compilation rules.
 #
-override _R_CFLAGS=$(CPPFLAGS) $(CFLAGS) $(INCFLAGS) -MD
-override _R_CXXFLAGS=$(CPPFLAGS) $(CFLAGS) $(CXXFLAGS) $(INCFLAGS) -MD
+override _R_CFLAGS=$(CPPFLAGS) $(CFLAGS) $(INCFLAGS) -MMD
+override _R_CXXFLAGS=$(CPPFLAGS) $(CFLAGS) $(CXXFLAGS) $(INCFLAGS) -MMD
 export _R_CFLAGS
 export _R_CXXFLAGS
 
@@ -85,7 +85,7 @@
 	@rm -f .$*.d $@
 	$(CXX) $$_R_CXXFLAGS -c $<
 	@mv $*.d .$*.d
-	
+
 ../%.a:
 	@echo "Library $@ does not exist!"; exit 1
 
diff -uNr wvdial-1.41/streams/pipetest.cc wvdial-libs/streams/pipetest.cc
--- wvdial-1.41/streams/pipetest.cc	Wed Oct 20 16:20:13 1999
+++ wvdial-libs/streams/pipetest.cc	Fri Jan 28 15:15:50 2000
@@ -12,7 +12,7 @@
 int main(int argc, char **argv)
 {
     const char *_av[] = {
-	"/bin/bash",
+       argc==2 ? argv[1] : "/bin/sh",
 	NULL
     };
     const char **av = (argc < 2) ? _av : (const char **)(argv + 1);
diff -uNr wvdial-1.41/streams/wvstream.cc wvdial-libs/streams/wvstream.cc
--- wvdial-1.41/streams/wvstream.cc	Wed Oct 20 18:37:04 1999
+++ wvdial-libs/streams/wvstream.cc	Fri Jan 28 15:15:50 2000
@@ -59,7 +59,7 @@
     char buf[1024];
     size_t len;
     
-    while (s.isok() && s.select(0))
+    while (s.isok() && s2.isok() && s.select(0))
     {
 	len = s.read(buf, sizeof(buf));
 	s2.write(buf, len);
diff -uNr wvdial-1.41/wvdial/wvdial.cc wvdial-libs/wvdial/wvdial.cc
--- wvdial-1.41/wvdial/wvdial.cc	Sun Nov 28 14:39:36 1999
+++ wvdial-libs/wvdial/wvdial.cc	Fri Jan 28 15:15:50 2000
@@ -90,6 +90,7 @@
     char *		homedir = getenv("HOME");
 
     bool chat_mode = false;
+    char *remote_name = NULL;
     
     signal( SIGTERM, signalhandler );
     signal( SIGINT, signalhandler );
@@ -109,6 +110,11 @@
 
     if( argc > 1 ) {
     	for( int i=1; i < argc; i++ ) {
+            if( !strcmp( argv[i], "--remotename" ) ) {
+                i++;
+                remote_name = argv[i];
+                continue;
+            }
 	    if( !strcmp( argv[i], "--chat" ) ) {
 		syslog = new WvSyslog( "WvDial", false, WvLog::Debug2, 
 				       WvLog::Debug2 );
@@ -133,7 +139,7 @@
 	sections->append( new WvString( "Dialer Defaults" ), true);
     }
     
-    WvDialer dialer( cfg, sections, chat_mode );
+    WvDialer dialer( cfg, sections, chat_mode, remote_name );
 	
     if( dialer.dial() == false )
 	return  1;
diff -uNr wvdial-1.41/wvdial/wvdialer.cc wvdial-libs/wvdial/wvdialer.cc
--- wvdial-1.41/wvdial/wvdialer.cc	Fri Nov 19 00:14:46 1999
+++ wvdial-libs/wvdial/wvdialer.cc	Fri Jan 28 15:15:50 2000
@@ -51,7 +51,7 @@
 //       WvDialer Public Functions
 //**************************************************
 
-WvDialer::WvDialer( WvConf &_cfg, WvStringList *_sect_list, bool _chat_mode = false )
+WvDialer::WvDialer( WvConf &_cfg, WvStringList *_sect_list, bool _chat_mode = false, char *_remote_name = NULL )
 /********************************************************/
 : WvStreamClone( (WvStream **)&modem ),
     cfg(_cfg), log( "WvDial", WvLog::Debug ),
@@ -73,6 +73,7 @@
     
     sect_list = _sect_list;
     chat_mode = _chat_mode;
+    remote_name = _remote_name;
     
     log( "WvDial: Internet dialer version " WVDIAL_VER_STRING "\n" );
 
@@ -408,6 +409,11 @@
     	    		    cfg.get( d, opts[i].name, opts[i].str_default ) );
     	}
     }
+
+    if( remote_name ) {
+	// command line overrides configuration file
+	options.remote = remote_name;
+    }
 }
 
 bool WvDialer::init_modem()
@@ -601,8 +607,10 @@
 	"-detach",
 	"user", options.login,
 	options.force_addr[0] ? (const char *)addr_colon : "noipdefault",
-	options.new_pppd ? "call" : NULL, 
-	options.new_pppd ? "wvdial" : NULL,
+	options.remote ? "remotename" : (options.new_pppd ? "call" : NULL), 
+	options.remote ? (char*)options.remote : (options.new_pppd ? "wvdial" : NULL),
+	(options.new_pppd && options.remote) ? "call" : NULL,
+	(options.new_pppd && options.remote) ? "wvdial" : NULL,
 	NULL
     };
 
diff -uNr wvdial-1.41/wvdial/wvdialer.h wvdial-libs/wvdial/wvdialer.h
--- wvdial-1.41/wvdial/wvdialer.h	Sat Aug  7 13:06:16 1999
+++ wvdial-libs/wvdial/wvdialer.h	Fri Jan 28 15:15:50 2000
@@ -42,7 +42,7 @@
 /***********************************/
 {
 public:
-    WvDialer( WvConf &_cfg, WvStringList *_sect_list, bool _chat_mode = false );
+    WvDialer( WvConf &_cfg, WvStringList *_sect_list, bool _chat_mode = false, char *_remote_name = NULL );
     virtual ~WvDialer();
 
     WvModemBase *modem;
@@ -115,6 +115,7 @@
     WvStringList *	sect_list;
 
     bool		chat_mode;
+    char *		remote_name;
     
     bool		been_online;
     time_t		connected_at;
diff -uNr wvdial-1.41/wvdial/wvdialtext.cc wvdial-libs/wvdial/wvdialtext.cc
--- wvdial-1.41/wvdial/wvdialtext.cc	Sun Aug  1 12:00:38 1999
+++ wvdial-libs/wvdial/wvdialtext.cc	Fri Jan 28 15:15:50 2000
@@ -11,12 +11,13 @@
 
 const char wvdial_version_text[] = 
 "WvDial " WVDIAL_VER_STRING
-": Copyright (c) 1998, 1999 Worldvisions Computer Technology, Inc.\n";
+": Copyright (c) 1998 - 2000 Worldvisions Computer Technology, Inc.\n";
 
 const char wvdial_help_text[] = 
 "Usage: wvdial { option || sect1 sect2 sect3 ... } \n"
 "\n"
 "  options:	--chat		used when running wvdial from pppd\n"
+"  		--remotename	remotename to pass to pppd\n"
 "  		--help		display this help and exit\n"
 " 		--version	output version information and exit\n"
 "\n"
diff -uNr wvdial-1.41/wvdial/wvmodemscan.h wvdial-libs/wvdial/wvmodemscan.h
--- wvdial-1.41/wvdial/wvmodemscan.h	Tue Mar 23 22:41:42 1999
+++ wvdial-libs/wvdial/wvmodemscan.h	Fri Jan 28 15:15:37 2000
@@ -45,6 +45,8 @@
     bool isdone() const
 	{ return stage == Done; }
     bool isok() const;
+    enum Stage get_stage() const
+        { return (enum Stage) stage; }
 
     // is this an isdn modem?  Returns modem identifier if so.
     const char *is_isdn() const;
