--- wvdial-1.41/wvdial/wvdialer.cc	Sun May 14 22:46:15 2000
+++ wvdial-1.41-p/wvdial/wvdialer.cc	Sun May 14 22:45:29 2000
@@ -608,7 +608,7 @@
 	"user", options.login,
 	options.force_addr[0] ? (const char *)addr_colon : "noipdefault",
 	options.remote ? "remotename" : (options.new_pppd ? "call" : NULL), 
-	options.remote ? (char*)options.remote : (options.new_pppd ? "wvdial" : NULL),
+	options.remote ? (const char*)options.remote : (options.new_pppd ? "wvdial" : NULL),
 	(options.new_pppd && options.remote) ? "call" : NULL,
 	(options.new_pppd && options.remote) ? "wvdial" : NULL,
 	NULL
